src/vanilla.ts

```diff
@@ -73,7 +73,7 @@ const createStoreImpl: CreateStoreImpl = (createState) => {
     if (!Object.is(nextState, state)) {
       const previousState = state
       state =
-        replace ?? typeof nextState !== 'object'
+        replace ?? (typeof nextState !== 'object' || nextState === null)
```

Comment:
This is what we agreed, right?
```suggestion
        replace ?? (typeof state !== 'object' || state === null)
```